reference/forms/types/date.rst

```diff
@@ -69,6 +69,14 @@ field as **three different choice fields**::
 If your underlying date is *not* a ``DateTime`` object (e.g. it's a unix timestamp),
 configure the `input`_ option.
 
+When working with a ``DateTimeImmutable`` object, add this as an ``input`` option::
```

Comment:
I don't think we should add a full new paragraph for it. We can update the paragraph above like (e.g. it is `DateTimeImmutable` object or a UNIX timestamp).